collections-arrays/ConvertHexStringToByteArray/src/ConvertHexStringToByteArrayLibrary/ConversionHelpers.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Runtime.CompilerServices;
+
+namespace ConvertHexStringToByteArray.Library;
+
+public static class ConversionHelpers
+{
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static int PerformModularArithmeticCalculation(char value) => (value % 32 + 9) % 25;
+
+    public static byte[] FromHexWithModularArithmetic(ReadOnlySpan<char> input)
+    {
+        if (input.StartsWith("0x"))
+            input = input[2..];
+
+        if (input.IsEmpty)
+            return Array.Empty<byte>();
+
+        if (input.Length % 2 != 0)
+            throw new ArgumentException("Input has invalid length", nameof(input));
+
+        var dest = new byte[input.Length >> 1];
+        for (int i = 0, j = 0; j < dest.Length; i += 2, j++)
```

Comment:
@AndreCarvalho - it can't be made into `var`, it is intentionally `int` because there are multiple declarators. The line is:
`int i=0, j=0`, which cannot be written with `var`